cmake/modules/contrib/ExampleTargetHooks.cmake

```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+file(GLOB EXAMPLE_TARGET_HOOKS_SRC src/relay/backend/contrib/example_target_hooks/relay_to_tir.cc)
```

Comment:
everything in this PR looks great, except that i'm a little concerned we're linking in test-only C++ here into `libtvm.so`. Possible to do this with TVMScript in Python? i feel like we would need to e.g. add a tests/libtest/src/*.cc plus a separate `cmake` build target to create a `.so` for the code in there that links against `libtvm.so`, then a pytest fixture to load it in one time at the start of testing and provide the module to tests for use. and that sounds like a lot of extra ask for this pr :/

Reply:
cc @jroesch @tqchen @junrushao1994

Reply:
it looks like there's precedent for this: https://github.com/apache/tvm/commit/c44b7bf107637ca81cd93eaee13e14a5e60386d5/cmake/modules/contrib/CODEGENC.cmake

so let's not block this PR on that. we will likely need to come to a solution for this in the future.